### PR TITLE
add dev container

### PR DESCRIPTION
**Pull request by joshspicer**
Adds a standard dev container configuration, setting up this repo for a great experience with VS Code Remote Containers, GitHub Codespaces, and any other implementor of the [dev container specification](https://containers.dev).

Configures a container with:
 - Go 1.18 installed with the appropriate flags
 - Docker running (inside the container, thus "docker-in-docker")

Additionally, if the user opens this dev container in VS Code, the official Go extension will be automatically installed.

### CODE REVIEW

**Comment by shizhMSFT** [comment]
@FeynmanZhou This PR is trying to merge IDE specific configurations to this repo. Do we have guidelines for this kind of operations?

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+ARG VARIANT="1.18"
+FROM mcr.microsoft.com/vscode/devcontainers/go:0-${VARIANT}
```
Comment (srivatsn) [inline .devcontainer/Dockerfile:2]: Can this use the universal image instead?

**Comment by FeynmanZhou** [comment]
> @FeynmanZhou This PR is trying to merge IDE specific configurations to this repo. Do we have guidelines for this kind of operations?

@shizhMSFT Not yet. 

I took a quick look at [dev container specification](https://containers.dev/) and want to figure out how can dev container help the ORAS developers to simplify the development process. Will it provide a fully remote development environment out-of-box running in a Docker container with Golang and other dependencies installed?

**Comment by sajayantony** [comment]
We would also need to continue to maintain this as the project evolves for e.g. go-lang updates. However as per precedence we avoided merging IDE specific changes into the repo and would recommend closing this for now.

**Comment by shizhMSFT** [comment]
Closing as IDE specific changes should be avoided.

**Pull request closed without merging**